index.js

```diff
@@ -0,0 +1,369 @@
+    remoteUri: internals.SCHEMA_REMOTE_URI,
+    timeout: internals.SCHEMA_TIMEOUT
+}).required().label('Remote Config');
+
+internals.SCHEMA_KEYS = Joi.alternatives(
+    internals.SCHEMA_LOCAL_CONFIG,
+    internals.SCHEMA_REMOTE_CONFIG
+).required().label('List of private keys or Remote URI for public keys');
+
+internals.SCHEMA_CREATE = Joi.object({
+    subject: Joi.string()
+        .required()
+        .label('Subject of the JWT'),
+    entity: Joi.object()
+        .required()
+        .label('Describing attributes about the subject'),
+    scope: Joi.array()
+        .default([])
+        .label('List of permissions applied to this JWT'),
+    time: Joi.string()
+        .required()
+        .label('Life of JWT')
+}).required();
+
+/**
+ * Creates a Source-of-Truth auth services
+ * @method loadLocal
+ * @param  {Object}  authConfig         Input for setting up a local version
+ * @param  {Array}   authConfig.keys    List of public/private keys
+ * @param  {String}  authConfig.maxAge  Maximum age of JWTs
+ * @return {Object}                     API to expose and getKey function
+ */
+internals.loadLocal = (authConfig) => {
+    const keys = {};
+
+    authConfig.keys.forEach((key) => {
+        keys[guid(key.public)] = key;
+    });
+
+    const expireIds = Object.keys(keys).sort((keyA, keyB) =>
+        keys[keyA].expires - keys[keyB].expires
+    );
+
+    return {
+        /**
+        * Load the key data
+        * @method getKey
+        * @param  {String}   keyId Key you are looking for
+        * @param  {Function} next  Function to call when done (error, key)
+        */
+        getKey: (keyId, next) => {
+            if (keys[keyId]) {
+                return next(null, keys[keyId]);
+            }
+
+            return next(new Error('Specified Key Id not found'));
+        },
+
+        /**
+         * Exposed API for the auth plugin
+         * @type {Object}
+         */
+        api: {
+            /**
+             * Create new JWT based on the available keys
+             * @method createJWT
+             * @param  {Object}  jwtConfig
+             * @param  {String}  jwtConfig.subject Thing that this represents
+             * @param  {Object}  jwtConfig.entity  Attributes about the subject
+             * @param  {Array}   jwtConfig.scope   List of privileges
+             * @param  {String}  jwtConfig.time    How long the credential will last
+             * @return {String}                    Signed JSON Web Token
+             */
+            createJWT: (jwtConfig) => {
+                const parsedConfig = Joi.attempt(jwtConfig, internals.SCHEMA_CREATE,
+                    'Invalid JWT options');
+
+                if (ms(parsedConfig.time) > ms(authConfig.maxAge)) {
+                    throw new Error('Request exceeds max JWT age');
+                }
+
+                const signingKeyId = Object.keys(keys).find(keyId =>
+                    (ms(parsedConfig.time) + Date.now()) / 1000 < keys[keyId].expires
```

Comment:
does it just pick first non expired key ?

Reply:
Not really, it selects the first one that will expire AFTER the desired JWT expiration length.  Added comment to describe that flow.